src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py

```diff
@@ -331,6 +338,196 @@ def db_update(
     return instance
 
 
+#####
+#           sql server audit-policy & threat-policy
+#####
+
+
+# Finds a storage account's resource group by querying ARM resource cache.
+# Why do we have to do this: so we know the resource group in order to later query the storage API
+# to determine the account's keys and endpoint. Why isn't this just a command line parameter:
+# because if it was a command line parameter then the customer would need to specify storage
+# resource group just to update some unrelated property, which is annoying and makes no sense to
+# the customer.
+def _find_storage_account(name):
+    resource_type = 'Microsoft.Storage/storageAccounts'
```

Comment:
Azure CLI doesn't support creating classic storage accounts, so I have no way to create them for automated test. On top of that, there is no python client library, so I would need to handwrite http request - exactly the kind of thing you would want to have automated. The customer can still use classic storage by specifying storage endpoint and key instead of account name, so classic storage isn't totally blocked, just inconvenient. I'm ok with this for the first release. I'm adding explicit checks.